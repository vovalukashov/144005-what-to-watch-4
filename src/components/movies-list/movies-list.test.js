import React from "react";
import renderer from "react-test-renderer";
import MoviesList from "./movies-list.jsx";

const movies = [
  {
    name: `Сияние`,
    image: `https://birdinflight.imgix.net/wp-content/uploads/2018/06/shining-sequel-movie_cover.jpg?fm=pjpg&q=70&fit=crop&crop=faces&w=630&h=446`,
  },
  {
    name: `Берегись автомобиля`,
    image: `https://birdinflight.imgix.net/wp-content/uploads/2018/06/shining-sequel-movie_cover.jpg?fm=pjpg&q=70&fit=crop&crop=faces&w=630&h=446`,
  },
  {
    name: `Солярис`,
    image: `https://birdinflight.imgix.net/wp-content/uploads/2018/06/shining-sequel-movie_cover.jpg?fm=pjpg&q=70&fit=crop&crop=faces&w=630&h=446`,
  }
];

const handleMovieCardEnter = jest.fn();
const handleMovieTitleClick = jest.fn();

describe(`MoviesList snapshot`, () => {
  it(`MoviesList renderer`, () => {
    const tree = renderer
      .create(<MoviesList movies={movies} onMovieCardEnter={handleMovieCardEnter} onMovieTitleClick={handleMovieTitleClick} />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
