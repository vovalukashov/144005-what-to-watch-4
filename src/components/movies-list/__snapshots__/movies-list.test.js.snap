// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MoviesList snapshot MoviesList renderer 1`] = `
<div
  className="catalog__movies-list"
>
  <article
    className="small-movie-card catalog__movies-card"
    onMouseEnter={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <img
        alt="Fantastic Beasts: The Crimes of Grindelwald"
        height="175"
        src="https://birdinflight.imgix.net/wp-content/uploads/2018/06/shining-sequel-movie_cover.jpg?fm=pjpg&q=70&fit=crop&crop=faces&w=630&h=446"
        width="280"
      />
    </div>
    <h3
      className="small-movie-card__title"
      onClick={[MockFunction]}
    >
      <a
        className="small-movie-card__link"
        href="movie-page.html"
      >
        Сияние
      </a>
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
    onMouseEnter={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <img
        alt="Fantastic Beasts: The Crimes of Grindelwald"
        height="175"
        src="https://birdinflight.imgix.net/wp-content/uploads/2018/06/shining-sequel-movie_cover.jpg?fm=pjpg&q=70&fit=crop&crop=faces&w=630&h=446"
        width="280"
      />
    </div>
    <h3
      className="small-movie-card__title"
      onClick={[MockFunction]}
    >
      <a
        className="small-movie-card__link"
        href="movie-page.html"
      >
        Берегись автомобиля
      </a>
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
    onMouseEnter={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <img
        alt="Fantastic Beasts: The Crimes of Grindelwald"
        height="175"
        src="https://birdinflight.imgix.net/wp-content/uploads/2018/06/shining-sequel-movie_cover.jpg?fm=pjpg&q=70&fit=crop&crop=faces&w=630&h=446"
        width="280"
      />
    </div>
    <h3
      className="small-movie-card__title"
      onClick={[MockFunction]}
    >
      <a
        className="small-movie-card__link"
        href="movie-page.html"
      >
        Солярис
      </a>
    </h3>
  </article>
</div>
`;
